src/openms/include/OpenMS/CONCEPT/Colorizer.h

```diff
@@ -0,0 +1,248 @@
+   *
+   */
+  class Colorizer
+  {
+
+  public:
+    /** @name Constructors and Destructor
+     */
+    //@{
+    Colorizer(int color); 
+
+    /// Copy constructor
+    //Colorizer(const Colorizer &rhs);
+
+    /// Default destructor
+    ~Colorizer();
+    //@}
+
+    
+    auto getColor(int i = -1);
+    std::string getText();
+    bool getReset() {return this->reset;}
+
+//operator overloading
+    friend std::ostream &operator<<(std::ostream &o_stream, Colorizer& col);
+    
+    
+    
+    template <typename T = std::string>
+    Colorizer& operator()(T s = T(""))
+    {
+      //clear was not possible (resets some flags in the sream)
+      //std::string str = "";
+      this->_input.str("");
+      this->_input << s;
+     
+
+      (this->_input.str() == "")? this->reset = false: this->reset = true;
+      
+      return *this;
+    }
+    /*
+    Colorizer& operator()()
+    {
+      
+      this ->reset = false;
+      this->_input.str("");
+      return *this;
+    }
+    */
+
+/*
+//klammer operator, um die farbigen Text an eine funktion übergeben zu können. 
+    template <typename T>
+    friend std::string operator()(T s)
+    {
+
+      std::stringstream text;
+      text << this;
+      return text.str();  
+    }
+*/
+   
+
+  private:
+    //std::string _text;
+    const int _color;
+    std::stringstream _input;
+    bool reset = true;
+#if defined(_WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(_WIN64)
+    int _defcolor;
+#endif
+   //ewnum für farbauswahl
+
+
+
+/**
+     * @brief constant string array which saves the Linux color codes.
+     * 0=black
+     * 1=red
+     * 2=green
+     * 3=yellow
+     * 4=blue
+     * 5=magenta
+     * 6=cyan
+     * 7=white
+     * 8=default console color (reset)
+     *
+     */
+#if defined(__linux__) || defined(__OSX__)
+    inline static const std::array<const char*,9> colors {"\033[30m", "\033[31m", "\033[32m", "\033[33m", "\033[34m", "\033[35m", "\033[36m", "\033[37m", "\033[0m"};
+#elif defined(_WIN32) || defined(_WIN32) || defined(__WIN32__) || defined(_WIN64)
+    inline static const std::array<const int,9> colors {16,12,10,14,9,13,11,15,15};
+#endif
+  };
+
+
+//deklaration of all colorizer object. 
+  extern /*OPENMS_DLLAPI*/ Colorizer color_black;
+  extern /*OPENMS_DLLAPI*/ Colorizer make_red;
+  extern /*OPENMS_DLLAPI*/ Colorizer make_green;
+  extern /*OPENMS_DLLAPI*/ Colorizer make_yellow;
+  extern /*OPENMS_DLLAPI*/ Colorizer make_blue;
+  extern /*OPENMS_DLLAPI*/ Colorizer make_magenta;
+  extern /*OPENMS_DLLAPI*/ Colorizer make_cyan;
+  extern /*OPENMS_DLLAPI*/ Colorizer make_white;
+  extern /*OPENMS_DLLAPI*/ Colorizer make_def;
+
+
+//definition of colorizing functions. (so function calls like process_string(green(string)) are possible.)
+  template <typename T = std::string>
+  extern std::string black(T s = T(""))
+  {
+    std::stringstream text;
+    text << color_black(s);
+      return text.str(); 
```

Comment:
```suggestion
      return color_black(s); 
```
same with the other ones

Reply:
doesn't word, bc. there is the return expects a std::string, the Object (color_black(s)) is an object of type colorizer, with no automatic cast to std::string. 
thats why I use the workaround, with the overloaded operator <<, so i can return a std::string object, casted from the  std::stringstream object.